vendor/desktop-notifications/README.md

```diff
@@ -0,0 +1,98 @@
+# desktop-notifications
+
+## A simple and opinionated library for working with OS notifications
+
+## Goals
+
+- zero dependencies
+- good support for Windows notifications
+- leverage TypeScript declarations wherever possible
+
+**Note:** This is currently in preview, with support for features that GitHub
+Desktop uses.
+
+## Install
+
+```shellsession
+$ npm install --save desktop-notifications
+# or
+$ yarn add desktop-notifications
+```
+
+## But Why?
+
+The current set of options for interacting with notifications, especially on
+Windows, have some limitations that meant we couldn't use them easily in GitHub
+Desktop:
+
+- [`electron`](https://www.electronjs.org/) doesn't support Windows
+  notifications when those are hidden away in the Action Center, because it
+  doesn't have a COM activator that could leverage CLSID-based activation. More
+  details about this can be found in
+  [electron/electron#29461](https://github.com/electron/electron/issues/29461).
+- [`node-notifier`](https://www.npmjs.com/package/node-notifier) relies on
+  [`snoretoast`](https://github.com/KDE/snoretoast) to handle notifications on
+  Windows, and the way it's used is only able to detect one event with each
+  notification, and also requires the app to use the same CLSID that is
+  [hardcoded](https://github.com/KDE/snoretoast/blob/17f88b2c757d54581bb7d5aa4d0d4462c3e75a98/CMakeLists.txt#L5)
+  in snoretoast.
+- [`electron-windows-notifications`](https://github.com/felixrieseberg/electron-windows-notifications)
+  has **many** dependencies around [`NodeRT`](https://github.com/NodeRT/NodeRT)
+  which, as of today, also require some
+  [manual steps](https://stackoverflow.com/a/54591996/673745) in order to build
+  them.
+
+After exploring all these options, we decided to write our own library to do the
+stuff we require using no dependencies at all and having all the features we
+need.
+
+## Building and running the sample app
+
+In order to get the sample app up and running you need to:
+
+1. Build `desktop-notifications`.
+2. Build the sample app.
+3. (macOS only) Sign the Electron binary, so that macOS will allow the sample
+   app to display notifications.
+4. Start the sample app.
```

Comment:
Do we need to bundle the sample app now that we're not advertising this for external use? Isn't Desktop sufficent?

Reply:
Hmmm maybe it is… truth is I used it to test the library as I worked on it. I guess we can now use Desktop for that 👍

Reply:
Removed everything in [e0d8cb6](https://github.com/desktop/desktop/pull/20248/commits/e0d8cb6d55d1ed1e349682b78c1f308e326aacc0)